tests/test_spe.py

```diff
@@ -1,51 +1,53 @@
 from datetime import datetime
+import os
+import shutil
+import struct
 
 import numpy as np
 import pytest
 
-import imageio.v2 as iio
+import imageio.v3 as iio
 from imageio.plugins import spe
-from conftest import deprecated_test
 
 
-@deprecated_test
-def test_spe_format():
-    for name in ("spe", ".spe"):
-        fmt = iio.formats[name]
-        assert isinstance(fmt, spe.SpeFormat)
+frame0 = np.zeros((32, 32), np.uint16)
+frame1 = np.ones_like(frame0)
 
 
-def test_spe_reading(test_images):
-    fname = test_images / "test_000_.SPE"
+def test_imopen(test_images, tmp_path):
+    with iio.imopen(test_images / "test_000_.SPE", "r") as img:
+        assert isinstance(img, spe.SpePlugin)
+
+    with iio.imopen(tmp_path / "test.spe", "w") as img:
+        assert not isinstance(img, spe.SpePlugin)
```

Comment:
```suggestion
    with pytest.raises(OSError):
        iio.imopen(tmp_path / "test.spe", "w", plugin="SPE")
```

Instead of testing that another plugin is found (which could fail) we should instead test that SPE declines to read SPE files. 

This is also in preparation for a feature on my wishlist, which is that plugin tests should run isolated from other plugins, i.e., if we run the tests for SPE then all other plugins should be disabled to avoid accidentally testing a different plugin. (The tested plugin might fail and ImageIO has logic in place to (sometimes) fall back on a different one that is known to be able to read this file type if possible.)